packages/at_client/lib/src/at_collection/at_collection_spec.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:at_client/at_client.dart';
+import 'package:at_client/src/at_collection/at_collection_impl.dart';
+import 'package:at_client/src/at_collection/model/at_operation_item_status.dart';
+
+abstract class AtCollectionSpec<T extends AtCollectionModel> {
+  /// Saves the object [T] in a self key.
+  /// A unique id is generated and stored in [T.id].
+  ///
+  /// Returns true if operation was successful else returns false.
+  ///
+  /// [expiryTime] Represents the time in milliseconds beyond which the key expires.
+  Future<bool> save(T model, {int? expiryTime});
+
+  /// updates object [T] using [T.id] as identifier.
+  /// Throws [Exception], if [T.id] or [T.collectionName] is empty.
+  ///
+  /// [update] also takes care of updating all the associated data with [T.id] if it has been shared with mutiple atSigns.
+  ///
+  /// returns List<AtOperationItemStatus>, where AtOperationItemStatus represents update status of individual key.
+  Future<List<AtOperationItemStatus>> update(T model, {int? expiryTime});
```

Comment:
Is it right to say that, update updates the self key and reshares it every one with whom it was previously shared?